code/modules/hydroponics/hydroponics_mutations.dm

```diff
@@ -0,0 +1,529 @@
+				weed_tolerance  =           max(0,  min(10,  weed_tolerance       + (rand(-2,2)   * degree)))
+				if(prob(degree*5))          parasite = !parasite
+
+			if(7)
+				lifespan =                  max(10, min(30,  lifespan             + (rand(-2,2)   * degree)))
+				if(yield != -1) yield =     max(0,  min(10,  yield                + (rand(-2,2)   * degree)))
+			if(8)
+				endurance =                 max(10, min(100, endurance            + (rand(-5,5)   * degree)))
+				production =                max(1,  min(10,  production           + (rand(-1,1)   * degree)))
+				potency =                   max(0,  min(200, potency              + (rand(-20,20) * degree)))
+				if(prob(degree*5))
+					spread =                max(0,  min(2,   spread               + rand(-1,1)))
+					source_turf.visible_message("<span class='notice'>\The [display_name] spasms visibly, shifting in the tray.</span>")
+			if(9)
+				maturation =                max(0,  min(30,  maturation      + (rand(-1,1)   * degree)))
+				if(prob(degree*5))
+					harvest_repeat = !harvest_repeat
+			if(10)
+				if(prob(degree*4))
+					biolum = !biolum
+					if(biolum)
+						source_turf.visible_message("<span class='notice'>\The [display_name] begins to glow!</span>")
+						if(prob(degree*4))
+							biolum_colour = "#[pick(list("FF0000","FF7F00","FFFF00","00FF00","0000FF","4B0082","8F00FF"))]"
+							source_turf.visible_message("<span class='notice'>\The [display_name]'s glow <font color='[biolum_colour]'>changes colour</font>!</span>")
+					else
+						source_turf.visible_message("<span class='notice'>\The [display_name]'s glow dims...</span>")
+			if(11)
+				if(prob(degree*2))
+					flowers = !flowers
+					if(flowers)
+						source_turf.visible_message("<span class='notice'>\The [display_name] sprouts a bevy of flowers!</span>")
+						if(prob(degree*2))
+							flower_colour = "#[pick(list("FF0000","FF7F00","FFFF00","00FF00","0000FF","4B0082","8F00FF"))]"
+						source_turf.visible_message("<span class='notice'>\The [display_name]'s flowers <font='[flower_colour]'>changes colour</font>!</span>")
+					else
+						source_turf.visible_message("<span class='notice'>\The [display_name]'s flowers wither and fall off.</span>")*/
+
+//Returns a key corresponding to an entry in the global seed list.
+/datum/seed/proc/get_mutant_variant()
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/datum/seed/proc/get_mutant_variant() called tick#: [world.time]")
+	if(!mutants || !mutants.len || immutable > 0) return 0
+	return pick(mutants)
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/mutate_species()
+
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/obj/machinery/portable_atmospherics/hydroponics/proc/mutate_species() called tick#: [world.time]")
+
+	var/previous_plant = seed.display_name
+	var/newseed = seed.get_mutant_variant()
+	if(newseed in plant_controller.seeds)
+		seed = plant_controller.seeds[newseed]
+	else
+		return
```

Comment:
You can change the above block of code:
``` DM
	if(newseed in plant_controller.seeds)
		seed = plant_controller.seeds[newseed]
	else
		return
```
Into:
``` DM
	if(!plant_controller.seeds.Find(newseed))
		return

	seed = plant_controller.seeds[newseed]
```

It's technically a minor nitpick because I feel an else for a bare return is useless, also the conversion of `in` to `Find()` is better for performance.